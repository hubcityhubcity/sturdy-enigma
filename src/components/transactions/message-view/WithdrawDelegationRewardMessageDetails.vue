<template>
  <div>
    <div class="tx__content__caption">
      <p>Withdrawal</p>
    </div>
    <div class="tx__content__information">
      From&nbsp;
      <router-link
        :to="`staking/validators/${transaction.value.validator_address}`"
        >{{
          transaction.value.validator_address | resolveValidatorName(validators)
        }}</router-link
      >
    </div>
  </div>
</template>

<script>
import { atoms, viewDenom, prettyLong } from "scripts/num.js"
import { resolveValidatorName } from "src/filters"

export default {
  name: `withdraw-delegation-reward-message-details`,
  filters: {
    atoms,
    viewDenom,
    prettyLong,
    resolveValidatorName
  },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Object,
      required: true
    }
  }
}
</script>
