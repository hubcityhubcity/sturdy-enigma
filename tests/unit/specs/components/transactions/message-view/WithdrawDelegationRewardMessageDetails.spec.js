import { shallowMount } from "@vue/test-utils"
import WithdrawDelegationRewardMessageDetails from "src/components/transactions/message-view/WithdrawDelegationRewardMessageDetails"

describe(`WithdrawalDelegationRewardMessageDetails`, () => {
  let wrapper
  // Examples @ https://stargate.cosmos.network/txs?action=withdraw_validator_rewards_all
  const tx = {
    height: 2000000,
    value: {
      delegator_address: "cosmos15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2kc6z4s",
      validator_address: "cosmosvaloper15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2nvwher"
    },
    fee: {
      amount: "37",
      denom: "uatom"
    },
    hash: `ABCD1234`,
    group: `governance`,
    key: "keyhash",
    memo: "(Sent via Lunie)",
    timestamp: 123456789,
    liquidDate: null,
    type: `cosmos-sdk/MsgWithdrawDelegationReward`
  }

  const validators = {
    cosmosvaloper15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2nvwher: {
      name: `SuperValidator`,
      operatorAddress: `cosmosvaloper15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2nvwher`,
      picture: "",
      __typename: `Validator`
    },
    cosmos1: {
      name: `SuperValidator`,
      operatorAddress: `"cosmos1"`,
      picture: "",
      __typename: `Validator`
    }
  }

  it(`renders a withdraw delegation reward transaction message`, () => {
    wrapper = shallowMount(WithdrawDelegationRewardMessageDetails, {
      propsData: {
        transaction: tx,
        validators: validators
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`renders a withdraw delegation reward transaction message without validator`, () => {
    wrapper = shallowMount(WithdrawDelegationRewardMessageDetails, {
      propsData: {
        transaction: tx,
        validators: validators
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
