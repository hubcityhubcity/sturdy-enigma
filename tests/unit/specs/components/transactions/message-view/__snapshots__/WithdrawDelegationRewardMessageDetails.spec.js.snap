// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithdrawalDelegationRewardMessageDetails renders a withdraw delegation reward transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="governance"
    transactiontype="Claimed"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Claimed
    </h3>
     
    <span>
      Rewards from 
    </span>
     
    <router-link-stub
      class="validator-link"
      to="/staking/validators/cosmosvaloper15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2nvwher"
    >
      <!---->
      
      SuperValidator
    
    </router-link-stub>
  </div>
</div>
`;

exports[`WithdrawalDelegationRewardMessageDetails renders a withdraw delegation reward transaction message without validator 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="governance"
    transactiontype="Claimed"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Claimed
    </h3>
     
    <span>
      Rewards from 
    </span>
     
    <router-link-stub
      class="validator-link"
      to="/staking/validators/cosmosvaloper15r4tc0m6hc7z8drq3dzlrtcs6rq2q9l2nvwher"
    >
      <!---->
      
      SuperValidator
    
    </router-link-stub>
  </div>
</div>
`;
